server/command.js

```diff
@@ -0,0 +1,243 @@
+$.utils.command.prepositions['on top of'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['on'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['onto'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['upon'] = 'on top of/on/onto/upon';
+$.utils.command.prepositions['over'] = 'over';
+$.utils.command.prepositions['through'] = 'through';
+$.utils.command.prepositions['under'] = 'under/underneath/beneath';
+$.utils.command.prepositions['underneath'] = 'under/underneath/beneath';
+$.utils.command.prepositions['beneath'] = 'under/underneath/beneath';
+$.utils.command.prepositions['behind'] = 'behind';
+$.utils.command.prepositions['beside'] = 'beside';
+$.utils.command.prepositions['for'] = 'for/about';
+$.utils.command.prepositions['about'] = 'for/about';
+$.utils.command.prepositions['is'] = 'is';
+$.utils.command.prepositions['as'] = 'as';
+$.utils.command.prepositions['off'] = 'off/off of';
+$.utils.command.prepositions['off of'] = 'off/off of';
+
+$.utils.command.prepositionsRegExp = new RegExp(
+    '(?:^|\\s)(with|using|upon|underneath|under|to|through|over|out +of|' +
+    'onto|on +top +of|on|off +of|off|is|into|inside|in +front +of|' +
+    'in|from +inside|from|for|beside|beneath|behind|at|as|about)(?:\\s|$)');
+
+$.utils.command.parse = function(command, user) {
+  // Parse a user's command into components:
+  // verb, argstr, args, dobjstr, dobj, prepstr, iobjstr, iobj
+  var args = command.split(/ +/);
+  var verb = args.shift();
+  if (!verb) {
+    return undefined;
+  }
+  var argstr = command.substring(command.indexOf(verb) + verb.length + 1);
+  var dobjstr = '';
+  var dobj = null;
+  var iobjstr = '';
+  var iobj = null;
+  var prepstr = '';
+  var m = argstr.match($.utils.command.prepositionsRegExp);
+  if (m) {
+    prepstr = m[1];
+    var i = argstr.indexOf(prepstr);
+    if (i === -1) {
+      throw RangeError('Impossible preposition: ' + prepstr);
+    }
+    dobjstr = argstr.substring(0, i - 1);
+    iobjstr = argstr.substring(i + prepstr.length + 1);
+    prepstr = prepstr.replace(/ +/g, ' ');
+  } else {
+    dobjstr = argstr;
+  }
+  if (dobjstr) {
+    var m = $.utils.match(dobjstr, user);
+    if (!m.length) {
+      dobj = undefined;
+    } else if (m.length === 1) {
+      dobj = dobj[0];
+    } else {
+      dobj = ReferenceError(dobj.length + ' matches.');
+    }
+  }
+  if (iobjstr) {
+    var m = $.utils.match(iobjstr, user);
+    if (!m.length) {
+      iobj = undefined;
+    } else if (m.length === 1) {
+      iobj = iobj[0];
+    } else {
+      iobj = ReferenceError(iobj.length + ' matches.');
+    }
+  }
+  return {
+    verb: verb,
+    argstr: argstr,
+    args: args,
+    dobjstr: dobjstr,
+    dobj: dobj,
+    prepstr: prepstr,
+    iobjstr: iobjstr,
+    iobj: iobj
+  };
+};
+
+$.utils.command.execute = function(command, user) {
+  // Parse and execute a user's command.
+  // Return true if successful, false if not.
+  var spec = $.utils.command.parse(command, user);
```

Comment:
Yeah: I did not mean to suggest overwriting command.  I was thinking more of what the canonical name of the parameter for verb functions should be, rather than of the local variable in this function.